### ISSUE TITLE
[Request Feature]: Not working in React 19

### ISSUE THREAD

**Issue by VichetN** [onesine/react-tailwindcss-datepicker#284]
![image](https://github.com/user-attachments/assets/f94f455a-8745-43c0-bd0c-a26da352c7a3)

Hope to have this amazing Datepicker compatable with React 19 soon :)

Or if this is a bug, please give me a solution to fix this Error.

**Comment by austingdev** [comment]
Similar error for me, using nextjs 15.0.1

![Screenshot 2024-10-23 at 10 02 04 AM](https://github.com/user-attachments/assets/0581de47-2c83-43bd-8ade-328b05e2e90f)

**Comment by carstenblt** [comment]
Actually just updating the react dependency works out of the box.

**Comment by austingdev** [comment]
@carstenblt  you are absolutely correct! I thought I had upgraded to React 19 but it appears I did not. 

For those coming here with the same problem, upgrading React to version `rc` (which ends up being 19) and installing fixed this problem for me.

**Comment by VichetN** [comment]
> @carstenblt you are absolutely correct! I thought I had upgraded to React 19 but it appears I did not.
> 
> For those coming here with the same problem, upgrading React to version `rc` (which ends up being 19) and installing fixed this problem for me.
> 
> @VichetN, give the upgrade a try to verify that fixed it for you. If so then closing this issue out would be ideal.

@austingdev How to upgrade by the way?

Upgrade React of our project?
should be like this?: 
`yarn add react@19.0.0-rc.0 react-dom@19.0.0-rc.0`

Please give me an example

**Comment by austingdev** [comment]
> > @carstenblt you are absolutely correct! I thought I had upgraded to React 19 but it appears I did not.
> > For those coming here with the same problem, upgrading React to version `rc` (which ends up being 19) and installing fixed this problem for me.
> > @VichetN, give the upgrade a try to verify that fixed it for you. If so then closing this issue out would be ideal.
> 
> @austingdev How to upgrade by the way?
> 
> Upgrade React of our Nextjs project? should be like this?: `yarn add react@19.0.0-rc.0 react-dom@19.0.0-rc.0`
> 
> Please give me an example

yes for the NextJS project. You can update the package.json libraries like so:
```
"react": "rc",
"react-dom": "rc",
```

**Comment by VichetN** [comment]
> > > @carstenblt you are absolutely correct! I thought I had upgraded to React 19 but it appears I did not.
> > > For those coming here with the same problem, upgrading React to version `rc` (which ends up being 19) and installing fixed this problem for me.
> > > @VichetN, give the upgrade a try to verify that fixed it for you. If so then closing this issue out would be ideal.
> > 
> > 
> > @austingdev How to upgrade by the way?
> > Upgrade React of our Nextjs project? should be like this?: `yarn add react@19.0.0-rc.0 react-dom@19.0.0-rc.0`
> > Please give me an example
> 
> yes for the NextJS project. You can update the package.json libraries like so:
> 
> ```
> "react": "rc",
> "react-dom": "rc",
> ```
> 
> then run `yarn`

Thank you @austingdev

**Comment by carstenblt** [comment]
Whut, wait? You just changed the version from `19.0.0-rc-69d4b800-20241021` to `rc` and it worked? So maybe it works with a newer release of react?

I actually had to rebuilt `react-tailwindcss-datepicker` with react 19 and push it to npm.

**Comment by austingdev** [comment]
Oh I might have had some kind of cacheing issue going on. I thought it was working but I nuked my dist folder and rebuilt and yea, even with `rc` it is still broken, same error as above. 

Does that mean you have pushed a new version that should work with `react@rc`?

**Comment by carstenblt** [comment]
> Oh I might have had some kind of cacheing issue going on. I thought it was working but I nuked my dist folder and rebuilt and yea, even with `rc` it is still broken, same error as above.
> 
> Does that mean you have pushed a new version that should work with `react@rc`?

Well yes, to `@tarabao/react-tailwindcss-datepicker`. If you trust that...

**Comment by mattschutt** [comment]
Why is this closed?  I do not consider a 3rd party fixing an issue an uploading to a different npm package a fix for the root issue... 

Can we get a fix for React 19 RC?

Please reopen until fixed?

**Comment by austingdev** [comment]
@mattschutt  I would tend to agree. 

@VichetN  It should be reopened until release from @onesine is provided and works with React RC.

**Comment by jaedung** [comment]
Same for me

**Comment by davecarlson** [comment]
FYI this still existing in the stable react 19 release

**Comment by Hasnain74** [comment]
The Issue still exist for react 19

**Comment by codedhola** [comment]
Hoping this is fixed sooner

**Comment by onesine** [comment]
Hi @VichetN, @mattschutt, @austingdev  👋 ,
Sorry for the late reply.
I've just released a new version that supports react version 19. For more details, please consult [Release 1.7.4](https://github.com/onesine/react-tailwindcss-datepicker/releases/tag/v1.7.4). 

You can try it and give me feedback.
